jdr/WFLY-13539-galleon_layer_jdr_support.adoc

```diff
@@ -0,0 +1,131 @@
+= Galleon layer for JBoss Diagnostic Reporting support
+:author:            Yeray Borges
+:email:             yborgess@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+The JBoss Diagnostic Reporting (JDR) subsystem enables the gathering of diagnostic data to aid in troubleshooting. JBoss EAP subscribers can provide this information to Red Hat when requesting support.
+
+The diagnosis information is generated by using the following operation:
+
+[source,xml]
+----
+[standalone@localhost:9990 /] /subsystem=jdr:generate-jdr-report()
+----
+
+Users are also able to generate the diagnostic information via command line executing the `jdr.sh` script located at `$WFLY_HOME/bin/` directory. This script accepts the following options:
+
+[source,bash]
+----
+$WFLY_HOME/bin/jdr.sh --help
+usage: jdr.{sh,bat,ps1} [options]
+
+JBoss Diagnostic Reporter (JDR) is a subsystem built to collect information to aid in troubleshooting. The jdr script is a utility for generating JDR reports.
+-c,--config <arg>     Configuration file of the server if it is not running.
+-h,--help             Display this message and exit
+-H,--host <arg>       hostname that the management api is bound to. (default: localhost)
+-p,--port <arg>       port that the management api is bound to. (default: 9990)
+-s,--protocol <arg>   Protocol that is used to connect. Can be remote, http or https (default: http)
+----
+
+The availability of a management interface is optional. If there is not a management interface available, the `jdr.{sh,bat,ps1}` script is still able to extract some information invoking internally the embedded server.
+
+The goal of this feature is to supply a https://docs.wildfly.org/galleon/#_layers[Galleon layer] to give support to the JDR subsystem.
+
+The following is a configuration example of the WildFly bootable jar maven plugin to provision a server with the `jdr` layer and the `core-tools` layer, which brings in the `jboss-cli.sh` script that can be used to connect to the server via the command line to generate a diagnostic report through the jdr subsystem:
```

Comment:
@yersan Do I understand correctly that `core-tools` layer here is optional and `jdr` would gather some report even without providing it?

Reply:
Hello @honza-kasik, right, `core-tools` is independent of `jdr`; is not part of the functionality `jdr` supplies. You can generate any report executing the`jdr.sh` shell scripts which are provisioned with `jdr` layer. 

This pattern is also followed by the other layers, `core-tools` is independent of any functionality supplied by a layer. If you need to connect to your server via CLI to do something, then you need to add it to your provisioning configuration. Remember it is included by default in all the three base layers.

I understand your question, the example added here is confusing about this spect. `jdr` is not meant to be used on community, is there to be used by the product. Normally, you will use `jdr` together with another decorator (jsf, ejb ...) and combined always with a base layer. So, product users will be able to either generate a report via `jdr` specific scripts or connecting to the server via command line and accessing to the `jdr` subsystem resource.

Reply:
Hi Yeray,

> I understand your question, the example added here is confusing about this spect. jdr is not meant to be used on community, is there to be used by the product. 

IIUC the Galleon layer will be available for WildFly as well, meaning that there will be a community version [1] of the artifact. Am I missing something?

[1]
https://issues.redhat.com/browse/WFLY-13539

Reply:
No, you are still on the track :), this layer is available on WildFly and user can use it, however, the utility of this layer is to supply aid information to GSS when you are creating a customer case, WildFly users don't create customer cases. A WildFly user can generate a report, that's correct, but the utility of this report for a community user is diffused.

Reply:
> No, you are still on the track :)

Great! ;)

> A WildFly user can generate a report, that's correct

That's enough for me to think that it is meaningful for the community as well and that the customer case scenario is not preventing this feature from being valuable for the end user (which could actually use the report on her/his own). 

Consequently IMHO it's safe to state that the code snippet is not bringing in any confusion about product/community versions.